### ISSUE TITLE
error at startup: some issue with magit-todos-keywords

### ISSUE THREAD

**Issue by benjaminor** [alphapapa/magit-todos#20]
I'm getting this error when trying to install magit-todos on emcas26.1:

I'm using the following code lines:
```
(use-package magit-todos
  :quelpa (magit-todos :fetcher github :repo "alphapapa/magit-todos")
  :config (magit-todos-mode))
```

Do you know what's causing the error?

> Debugger entered--Lisp error: (wrong-type-argument char-or-string-p nil)
>   rxt-parse-re(nil nil)
>   rxt-parse-elisp(nil)
>   rxt-elisp-to-pcre(nil)
>   #f(compiled-function (option value) #<bytecode 0x22e0fc1>)(magit-todos-keywords hl-todo-keyword-faces)
>   custom-reevaluate-setting(magit-todos-keywords)
>   #f(compiled-function (option value) #<bytecode 0x12e997d>)(magit-todos-ignored-keywords ("NOTE" "DONE"))
>   custom-initialize-reset(magit-todos-ignored-keywords (funcall (function #f(compiled-function () #<bytecode 0x1d21699>))))
>   custom-declare-variable(magit-todos-ignored-keywords (funcall (function #f(compiled-function () #<bytecode 0x1d21699>))) "Ignored keywords.  Automatically removed from `magit-todos-keywords'." :type (repeat string) :set #f(compiled-function (option value) #<bytecode 0x12e997d>))
>   byte-code("\300\301!\210\302\303\304\305\306\307%\210\310\311\312\313\314DD\315\316\317%\210\310\320\312\313\321DD\322\316\323\324\325&\007\210\310\326\312\313\327DD\330\316\331\324\332&\007\210\310\333\312\313\334DD\335\316\336\324\337&\007\210\310\340\312\313\341DD\342\316\343%\210\310\344\312\313\345DD\346\316\347%\210\310\350\312\313\351DD\352\316\353%\210\310\354\312\313\355DD\356\316\357%\210\310\360\312\313\361DD\362\316\317%\210\310\363\312\313\364DD\365\316\317%\210\310\366\312\313\367DD\370\316\371%\210\310\372\312\313\373DD\374\316\375%\210\310\376\312\313\377DD\201@\0\316\317%\210\310\201A\0\312\313\201B\0\ DD\201C\0\316\201D\0%\210\310\201E\0\312\313\201F\0\ DD\201G\0\316\201H\0%\210\310\201I\0\312\313\201J\0\ DD\201K\0\316\201L\0%\210\310\201M\0\312\313\201N\0\ DD\201O\0\316\201P\0%\207" [make-variable-buffer-local magit-todos-show-filenames custom-declare-group magit-todos nil "Show TODO items in source code comments in repos' files." :group magit custom-declare-variable magit-todos-fontify-keyword-headers funcall function #f(compiled-function () #<bytecode 0x1ea0f11>) "Apply keyword faces to group keyword headers." :type boolean magit-todos-ignored-keywords #f(compiled-function () #<bytecode 0x1d21699>) "Ignored keywords.  Automatically removed from `magit-todos-keywords'." (repeat string) :set #f(compiled-function (option value) #<bytecode 0x12e997d>) magit-todos-keywords #f(compiled-function () #<bytecode 0x11fa2a5>) "To-do keywords to display in Magit status buffer.\nIf set to a list variable, may be a plain list or an alist in\nwhich the keys are the keywords.\n\nWhen set, sets `magit-todos-search-regexp' to the appropriate\nregular expression." (choice (repeat :tag "Custom list" string) (const :tag "Keywords from `hl-todo'" hl-todo-keyword-faces) (variable :tag "List variable")) #f(compiled-function (option value) #<bytecode 0xcf86ed>) magit-todos-scan-fn #f(compiled-function () #<bytecode 0xcf6a59>) "File scanning method.\n\"Automatic\" will attempt to use rg, ag, git-grep, and\nfind-grep, in that order. " (choice (const :tag "Automatic" nil) (const :tag "rg" magit-todos--rg-scan-async) (const :tag "ag" magit-todos--ag-scan-async) (const :tag "git-grep" magit-todos--git-grep-scan-async) (const :tag "find-grep" magit-todos--grep-scan-async) (function :tag "Custom function")) #f(compiled-function (option value) #<bytecode 0x87d5b1>) magit-todos-max-items #f(compiled-function () #<bytecode 0xcf69d9>) "Automatically collapse the section if there are more than this many items." integer magit-todos-auto-group-items #f(compiled-function () #<bytecode 0xcf86b5>) "Whether or when to automatically group items." (choice (integer :tag "When there are more than this many items") (const :tag "Always" always) (const :tag "Never" never)) magit-todos-group-by #f(compiled-function () #<bytecode 0xcf8699>) "How to group items.\nOne or more attributes may be chosen, and they will be grouped in\norder." (repeat (choice (const :tag "By filename" magit-todos-item-filename) (const :tag "By keyword" magit-todos-item-keyword) (const :tag "By first path component" magit-todos-item-first-path-component))) magit-todos-ignore-file-suffixes #f(compiled-function () #<bytecode 0xcf867d>) "Ignore files with these suffixes." (repeat string) magit-todos-ignore-case #f(compiled-function () #<bytecode 0xcf84fd>) "Upcase keywords found in files.\nIf nil, a keyword like \"Todo:\" will not be shown.  `upcase' can\nbe a relatively expensive function, so this can be disabled if\nnecessary." magit-todos-fontify-org #f(compiled-function () #<bytecode 0xcf84e1>) "Fontify items from Org files as Org headings." magit-todos-sort-order #f(compiled-function () #<bytecode 0xcf8421>) "Order in which to sort items." (repeat (choice (const :tag "Keyword" magit-todos--sort-by-keyword) (const :tag "Filename" magit-todos--sort-by-filename) (const :tag "Buffer position" magit-todos--sort-by-position) (function :tag "Custom function"))) magit-todos-depth #f(compiled-function () #<bytecode 0xcf98b1>) "Maximum depth of files in repo working tree to scan for to-dos.\nDeeper scans can be slow in large projects.  You may wish to set\nthis in a directory-local variable for certain projects." (choice (const :tag "Repo root directory only" 0) integer) magit-todos-nice #f(compiled-function () #<bytecode 0xcfc051>) "Run scanner with \"nice\"." magit-todos-insert-at #f(compiled-function () #<bytecode 0xe9d909>) "Insert the to-dos section after this section in the Magit status buffer.\nSpecific sections may be chosen, using the first symbol returned\nby evaluating \"(magit-section-ident (magit-current-section))\"\nin the status buffer with point on the desired section,\ne.g. `recent' for the \"Recent commits\" section.  Note that this\nmay not work exactly as desired when the built-in scanner is\nused." (choice (const :tag "Top" top) (const :tag "Bottom" bottom) (const :tag "After untracked files" untracked) (const :tag "After unstaged files" unstaged) (symbol :tag "After selected section")) magit-todos-ag-args #f(compiled-function () #<bytecode 0xefd731>) "Extra arguments to pass to ag." (repeat string) magit-todos-rg-args #f(compiled-function () #<bytecode 0x10b29fd>) "Extra arguments to pass to rg." (repeat string) magit-todos-git-grep-args #f(compiled-function () #<bytecode 0x1184325>) "Extra arguments to pass to git-grep." (repeat string)] 8)
>   require(magit-todos nil t)
>   (not (require 'magit-todos nil t))
>   (if (not (require 'magit-todos nil t)) (display-warning 'use-package (format "Cannot load %s" 'magit-todos) :error))
>   (condition-case err (if (not (require 'magit-todos nil t)) (display-warning 'use-package (format "Cannot load %s" 'magit-todos) :error)) ((debug error) (funcall use-package--warning125 :catch err)))
>   (progn (use-package-ensure-elpa 'magit-todos 'nil 'nil) (if (and quelpa-use-package-inhibit-loading-quelpa (package-installed-p 'magit-todos)) nil (apply 'quelpa '((magit-todos :fetcher github :repo "alphapapa/magit-todos")))) (defvar use-package--warning125 (function (lambda (keyword err) (let ((msg (format "%s/%s: %s" 'magit-todos keyword (error-message-string err)))) (display-warning 'use-package msg :error))))) (condition-case err (if (not (require 'magit-todos nil t)) (display-warning 'use-package (format "Cannot load %s" 'magit-todos) :error)) ((debug error) (funcall use-package--warning125 :catch err))))
>   eval((progn (use-package-ensure-elpa 'magit-todos 'nil 'nil) (if (and quelpa-use-package-inhibit-loading-quelpa (package-installed-p 'magit-todos)) nil (apply 'quelpa '((magit-todos :fetcher github :repo "alphapapa/magit-todos")))) (defvar use-package--warning125 (function (lambda (keyword err) (let ((msg (format "%s/%s: %s" 'magit-todos keyword (error-message-string err)))) (display-warning 'use-package msg :error))))) (condition-case err (if (not (require 'magit-todos nil t)) (display-warning 'use-package (format "Cannot load %s" 'magit-todos) :error)) ((debug error) (funcall use-package--warning125 :catch err)))) nil)
>   elisp--eval-last-sexp(nil)
>   eval-last-sexp(nil)
>   funcall-interactively(eval-last-sexp nil)
>   call-interactively(eval-last-sexp nil nil)
>   command-execute(eval-last-sexp)

**Comment by benjaminor** [comment]
I installed magit-todos yesterday on a different machine and it worked just fine then.
The installed version was magit-todos-20180627.631.

The version that's causing the trouble is magit-todos-20180628.515.

**Comment by alphapapa** [comment]
Oops, I made a mistake when I modified Jiangbin's commit.  I just fixed it in commit 1cc137300a8a6231f257d71bbddea26dfa08884b.  Please update to the latest `master` and it should work again.

Thank you very much for promptly reporting this and with detail!
